packages/core/src/App/Containers/Layout/header/header.tsx

```diff
@@ -86,11 +91,13 @@ const Header = observer(() => {
             case pathname === routes.onboarding:
                 result = null;
                 break;
+            case is_dtrader_v2_enabled && is_mobile && pathname === routes.trade:
+                result = <DTraderV2Header />;
+                break;
             case is_dtrader_v2_enabled &&
                 is_mobile &&
-                (pathname.startsWith(routes.trade) ||
-                    pathname.startsWith('/contract/') === routes.contract.startsWith('/contract/')):
-                result = <DTraderV2Header />;
+                pathname.startsWith('/contract/') === routes.contract.startsWith('/contract/'):
```

Comment:
During testing I noticed weird header. Sometimes we are showing causual header, sometimes dtrader-v2 header, sometimes header for contract details
https://github.com/binary-com/deriv-app/assets/121025168/b5e9532a-50f6-4934-86a7-2bec1f3010be

Reply:
Maybe it will be better use 1 header file for Dtrader-V2 and inside it determine what exactly we should render?

Reply:
+ I think this use-case check is not triggered when we navigate between pages. It's triggers ones and then only after refreshing the page.

Reply:
It's probably because of the lazy loading. Can we remove the lazy loading for the header component?

Reply:
Its fixed , thanks